File: HouseM8-Flutter/housem8_flutter/lib/models/user_register.dart
import 'package:housem8_flutter/models/address.dart';

class UserRegister {
  final String firstName;
  final String lastName;
  final String userName;
  final String password;
  final String email;
  final String description;
  final Address address;

  UserRegister(
      {this.firstName,
      this.lastName,
      this.userName,
      this.password,
      this.email,
      this.description,
      this.address});

  factory UserRegister.fromJson(Map<String, dynamic> json) {
    return UserRegister(
      firstName: json["firstName"],
      lastName: json["lastName"],
      userName: json["userName"],
      password: json["password"],
      email: json["email"],
      description: json["description"],
      address: Address.fromJson(json["address"]),
    );
  }
}

File: HouseM8API/HouseM8API/Models/Users/UserRegister.cs
using HouseM8API.Entities;
using System.ComponentModel.DataAnnotations;

namespace HouseM8API.Models
{
    /// <summary>
    /// Modelo usado por um utilizador
    /// para resgistar os dados pessoais
    /// </summary>
    public abstract class UserRegister
    {
        [Required]
        public string FirstName { get; set; }
        [Required]
        public string LastName { get; set; }
        [Required]
        public string UserName { get; set; }
        [Required]
        public string Password { get; set; }
        [Required]
        [EmailAddress]
        public string Email { get; set; }
        public string Description { get; set; }
        [Required]
        public Address Address { get; set; }
    }
}
